doc/source/building/index.rst

```diff
@@ -0,0 +1,354 @@
+
+          sudo apt build-dep scipy
+
+        This command installs whatever is needed to build SciPy, with the
+        advantage that new dependencies or updates to required versions are
+        handled by the package managers.
+
+      .. tab-item:: Fedora
+
+        To install SciPy build requirements, you can do::
+
+          sudo dnf install gcc-gfortran python3-devel openblas-devel lapack-devel pkgconfig
+
+        Alternatively, you can do::
+
+          sudo dnf builddep scipy
+
+        This command installs whatever is needed to build SciPy, with the
+        advantage that new dependencies or updates to required versions are
+        handled by the package managers.
+
+      .. tab-item:: CentOS/RHEL
+
+        To install SciPy build requirements, you can do::
+
+          sudo yum install gcc-gfortran python3-devel openblas-devel lapack-devel pkgconfig
+
+        Alternatively, you can do::
+
+          sudo yum-builddep scipy
+
+        This command installs whatever is needed to build SciPy, with the
+        advantage that new dependencies or updates to required versions are
+        handled by the package managers.
+
+      .. tab-item:: Arch
+
+        To install SciPy build requirements, you can do::
+
+          sudo pacman -S gcc-fortran openblas pkgconf
+
+  .. tab-item:: macOS
+
+    Install Apple Developer Tools. An easy way to do this is to
+    `open a terminal window <https://blog.teamtreehouse.com/introduction-to-the-mac-os-x-command-line>`_,
+    enter the command::
+
+        xcode-select --install
+
+    and follow the prompts. Apple Developer Tools includes Git, the Clang C/C++
+    compilers, and other development utilities that may be required.
+
+    Do *not* use the macOS system Python. Instead, install Python
+    with `the python.org installer <https://www.python.org/downloads/>`__ or
+    with a package manager like Homebrew, MacPorts or Fink.
+
+    The other system dependencies you need are a Fortran compiler, BLAS and
+    LAPACK libraries, and pkg-config. They're easiest to install with
+    `Homebrew <https://brew.sh/>`__::
+
+        brew install gfortran openblas pkg-config
+
+    .. note::
+
+        As of SciPy >=1.2.0, we do not support compiling against the system
+        Accelerate library for BLAS and LAPACK. It does not support a sufficiently
+        recent LAPACK interface. This is planned to change in 2023, because macOS
+        13.3 introduced a major upgrade to Accelerate which resolved all known
+        issues.
+
+  .. tab-item:: Windows
+
+    A compatible set of C, C++ and Fortran compilers is needed to build SciPy.
+    This is tricker on Windows than on other platforms, because MSVC does not
+    support Fortran, and gfortran and MSVC can't be used together. You will
+    need one of these sets of compilers:
+
+    1. MSVC + Intel Fortran (``ifort``)
+    2. Intel compilers (``icc``, ``ifort``)
+    3. Mingw-w64 compilers (``gcc``, ``g++``, ``gfortran``)
```

Comment:
Is there a recommendation for those who don't already have compilers and who can't easily make an informed decision?

Reply:
That's what I would have said. Consider listing it first and noting that it's recommended if the reader is unsure.

Reply:
> That's what I would have said. Consider listing it first and noting that it's recommended if the reader is unsure.

Agreed, doing this.

> The one potential hangup I see is putting it on the path. There is a note about the necessity of doing that, but can we mention the specific location or recommend how to look for it?

Mingw-w64 is already on the path if you use `choco`, and there's the recommendation for how to check in the `.. note::`: run `gfortran --version`.

Reply:
This is the problem with Windows: too many manual steps, shells with different behavior, etc. I don't think it can be described in more detail without it becoming a maintenance burden again, and it's impossible to be completely exhaustive. Windows users who struggle with these instructions are better off using WSL, then things are trivially easy.

Reply:
> Mingw-w64 is already on the path

I'll have to check on that. It might be visible from Powershell but I'm not sure if it is for other shells.

I agree that it's impossible to be exhaustive and keep it up to date. If it is not added to the path for all shells by default, the one easy thing we might do is mention that the default location of the compilers can be found in "SciPy GitHub Actions CI jobs for Windows", which is already linked.

Reply:
Thanks for reordering these. The one thing I'd suggest is to also reorder the tabs so that the MinGW-w64 instructions show by default.

Reply:
Good idea. I reordered that tab as well now, resolving.